lnd_test.go

```diff
@@ -2133,6 +2133,65 @@ func testNodeAnnouncement(net *networkHarness, t *harnessTest) {
 	}
 }
 
+func testNodeSignVerify(net *networkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	// bob should be able to verify alice's signature
+
+	aliceMsg := []byte("alice msg")
+
+	// alice: sign "alice msg"
+
+	sigReq := &lnrpc.SignMessageRequest{Msg: aliceMsg}
+	sigResp, err := net.Alice.SignMessage(ctxb, sigReq)
+	if err != nil {
+		t.Fatalf("SignMessage rpc call failed: %v", err)
+	}
+	aliceSig := sigResp.Signature
+
+	// bob: verify alice's signature -> should succeed
+
+	verifyReq := &lnrpc.VerifyMessageRequest{
+		Msg: aliceMsg, Signature: aliceSig}
+	verifyResp, err := net.Bob.VerifyMessage(ctxb, verifyReq)
+	if err != nil {
+		t.Fatalf("VerifyMessage failed: %v", err)
+	}
+	if !verifyResp.Valid {
+		t.Fatalf("alice's signature didn't validate")
+	}
+
+	// carol is a new node that is unconnected to alice or bob
+
+	carol, err := net.NewNode(nil)
+	if err != nil {
+		t.Fatalf("unable to create new node: %v", err)
+	}
+
+	carolMsg := []byte("carol msg")
+
+	// carol: sign "carol msg"
+
+	sigReq = &lnrpc.SignMessageRequest{Msg: carolMsg}
+	sigResp, err = carol.SignMessage(ctxb, sigReq)
+	if err != nil {
+		t.Fatalf("SignMessage rpc call failed: %v", err)
+	}
+	carolSig := sigResp.Signature
+
+	// bob: verify carol's signature -> should fail
+
+	verifyReq = &lnrpc.VerifyMessageRequest{
+		Msg: carolMsg, Signature: carolSig}
+	verifyResp, err = net.Bob.VerifyMessage(ctxb, verifyReq)
+	if err != nil {
+		t.Fatalf("VerifyMessage failed: %v", err)
+	}
+	if verifyResp.Valid {
+		t.Fatalf("carol's signature should not be valid")
+	}
+}
```

Comment:
The `carol` node should be shutdown at the end of this test as it's no longer of any use:
```go
if err := carol.shutdown()l err != nil {
    ... handle err... 
}
```